tests/Routing/RoutingRouteTest.php

```diff
@@ -943,6 +943,26 @@ public function testWherePatternsProperlyFilter()
         });
         $route->where('bar', '[0-9]+');
         $this->assertFalse($route->matches($request));
+
+        /*
+         * Conditional
+         */
+        $route = new Route('GET', '{subdomain}.awesome.test', function () {
+            //
+        });
+
+        $route->when(true, function ($route) {
+            $route->whereIn('subdomain', [
+                'one',
+                'two',
+            ]);
+        });
+
+        $request = Request::create('test.awesome.test', 'GET');
+        $this->assertFalse($route->matches($request));
+
+        $request = Request::create('one.awesome.test', 'GET');
+        $this->assertTrue($route->matches($request));
```

Comment:
This test doesn't actually test if `when()` works but if `matches()` works—this is counterintuitive to have in this PR